core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -0,0 +1,166 @@
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.metadata.tasks.Task.Status;
+import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.core.events.ScriptEvaluator;
+import com.netflix.conductor.core.execution.WorkflowExecutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.script.ScriptException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Manan
+ *
+ */
+public class DoWhile extends WorkflowSystemTask {
+
+	public DoWhile() {
+		super("DO_WHILE");
+	}
+
+	Logger logger = LoggerFactory.getLogger(DoWhile.class);
+
+	@Override
+	public void cancel(Workflow workflow, Task task, WorkflowExecutor executor) {
+		task.setStatus(Status.CANCELED);
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean execute(Workflow workflow, Task task, WorkflowExecutor provider) {
+		
+		boolean allDone = true;
+		boolean hasFailures = false;
+		StringBuilder failureReason = new StringBuilder();
+		List<WorkflowTask> loopOver = task.getWorkflowTask().getLoopOver();
+
+		for(WorkflowTask workflowTask : loopOver){
+			String refName = getTaskRefName(workflowTask, task);
+			Task loopOverTask = workflow.getTaskByRefName(refName);
+			if(loopOverTask == null){
+				//Task is not even scheduled yet
+				allDone = false;
+				break;
+			}
+			Status taskStatus = loopOverTask.getStatus();
+			hasFailures = !taskStatus.isSuccessful();
+			if(hasFailures){
+				failureReason.append(loopOverTask.getReasonForIncompletion()).append(" ");
+			}
+			task.getOutputData().put(workflowTask.getTaskReferenceName(), loopOverTask.getOutputData());
+			allDone = taskStatus.isTerminal();
+			if(!allDone || hasFailures){
+				break;
+			}
+		}
+		if (hasFailures) {
+			return markLoopTaskFailed(task, failureReason.toString());
+		} else if (!allDone) {
+			return false;
+		}
+		boolean shouldContinue = getEvaluatedCondition(task, workflow);
+		logger.debug("workflowid {} shouldcontinue {}",workflow.getWorkflowId(), shouldContinue);
+		if (shouldContinue){
+			return scheduleLoopTasks(task, workflow, provider);
+		} else {
+			return markLoopTaskSuccess(task, workflow);
+		}
+	}
+
+	private String getTaskRefName(WorkflowTask workflowTask, Task task) {
+		String refName = workflowTask.getTaskReferenceName();
+		if (task.getIteration()>0) {
+			refName +=  "_" + task.getIteration();
+		}
+		return refName;
+	}
+
+	Map<String, Object> getConditionInput(Task task, Workflow workflow) {
+		Map<String, Object> map = new HashMap<>();
+		List<WorkflowTask> loopOverTask = task.getWorkflowTask().getLoopOver();
+		for (WorkflowTask workflowTask : loopOverTask) {
+			Task temp = workflow.getTaskByRefName(getTaskRefName(workflowTask, task));
+			map.put(workflowTask.getTaskReferenceName(), temp.getOutputData());
```

Comment:
Input to a condition will still be original taskrefname only. So we will pick latest iteration output as input to condition.